tests/components/netatmo/test_climate.py

```diff
@@ -0,0 +1,854 @@
+"""The tests for the Netatmo climate platform."""
+from unittest.mock import Mock
+
+import pytest
+
+from homeassistant.components.climate import (
+    DOMAIN as CLIMATE_DOMAIN,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_PRESET_MODE,
+    SERVICE_SET_TEMPERATURE,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.components.climate.const import (
+    ATTR_HVAC_MODE,
+    ATTR_PRESET_MODE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_BOOST,
+)
+from homeassistant.components.netatmo import climate
+from homeassistant.components.netatmo.climate import (
+    NA_THERM,
+    NA_VALVE,
+    PRESET_FROST_GUARD,
+    PRESET_SCHEDULE,
+)
+from homeassistant.components.netatmo.const import (
+    ATTR_SCHEDULE_NAME,
+    SERVICE_SET_SCHEDULE,
+)
+from homeassistant.const import ATTR_ENTITY_ID, ATTR_TEMPERATURE
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+
+async def test_setup_no_data(hass, entry_error):
+    """Test setup with erroneous data."""
+    await hass.async_block_till_done()
+
+    assert (
+        hass.data["netatmo"][entry_error.entry_id]["netatmo_data_handler"].data[
```

Comment:
Don't assert integration details. Assert some core state or library calls.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations